File: api_nestjs_distributed_order_platform/src/application/port/payment-gateway.repository.ts
import { GatewayResult } from "src/shared/interfaces/gateway.interface";


export interface PaymentGatewayInterface {
    createCharge(input: {
      paymentId: string;
      orderId: string;
      amount: number;
      idempotencyKey: string;
    }): Promise<GatewayResult>;
    retrieveCharge(input: {
      idempotencyKey: string;
      externalId?: string;
    }): Promise<GatewayResult>;
}

File: api_java_distributed_order_platform/src/main/java/com/project/order/application/port/PaymentGatewayRepository.java
package com.project.order.application.port;

import com.project.order.shared.interfaces.GatewayResult;

public interface PaymentGatewayRepository {
    GatewayResult createCharge(String paymentId, String orderId, double amount, String idempotencyKey);
    GatewayResult retrieveCharge(String idempotencyKey, String externalId);
}
